src/test/java/org/broadinstitute/hellbender/tools/walkers/validation/ReferenceBlockConcordanceIntegrationTest.java

```diff
@@ -0,0 +1,436 @@
+        confidenceConcordanceMetrics.read(new FileReader(confidenceConcordanceHistogramFile.toFile()));
+        Assert.assertEquals(confidenceConcordanceMetrics.getNumHistograms(), 1);
+        Histogram<String> confidenceConcordanceHistogram = confidenceConcordanceMetrics.getHistogram();
+
+        confidenceConcordanceHistogram.values().forEach(bin -> {
+            String[] confidenceValues = bin.getId().split(",");
+            Assert.assertEquals(confidenceValues[0], confidenceValues[1]);
+        });
+    }
+
+    private Pair<File, File> writeTestGVCFs(List<TestReferenceBlockConcordanceVariant> truthVariants, List<TestReferenceBlockConcordanceVariant> evalVariants) throws Exception {
+        File truthFile = createTempFile("truth", ".gvcf");
+        FileWriter writer = new FileWriter(truthFile);
+        writer.write("##fileformat=VCFv4.2\n");
+        writer.write("##contig=<ID=test_contig,length=1000>\n");
+        writer.write("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tTESTSAMPLE\n");
+        for (TestReferenceBlockConcordanceVariant variant : truthVariants) {
+            writer.write(String.format("test_contig\t%s\t.\tA\t%s\t%s\t.\t%s\tGT:GQ\t%s:%s\n",
+                    variant.start,
+                    variant.getAltAllele(),
+                    variant.getAltAllele().equals("<NON_REF>") ? "." : "1",
+                    variant.getAltAllele().equals("<NON_REF>") ? String.format("END=%s", variant.getStop()) : ".",
+                    variant.getAltAllele().equals("<NON_REF>") ? "0/0" : "0/1",
+                    variant.getConfidence()));
+        }
+        writer.close();
+
+        File evalFile = createTempFile("eval", ".gvcf");
+        writer = new FileWriter(evalFile);
+        writer.write("##fileformat=VCFv4.2\n");
+        writer.write("##contig=<ID=test_contig,length=1000>\n");
+        writer.write("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tTESTSAMPLE\n");
+        for (TestReferenceBlockConcordanceVariant variant : evalVariants) {
+            writer.write(String.format("test_contig\t%s\t.\tA\t%s\t%s\t.\t%s\tGT:GQ\t%s:%s\n",
+                    variant.start,
+                    variant.getAltAllele(),
+                    variant.getAltAllele().equals("<NON_REF>") ? "." : "1",
+                    variant.getAltAllele().equals("<NON_REF>") ? String.format("END=%s", variant.getStop()) : ".",
+                    variant.getAltAllele().equals("<NON_REF>") ? "0/0" : "0/1",
+                    variant.getConfidence()));
+        }
+        writer.close();
+
+        return new Pair<>(truthFile, evalFile);
+    }
+
+    @DataProvider
+    public Object[][] provideSyntheticGVCFs() {
+        return new Object[][] {
+                // No non_ref blocks
+                {
+                        // Truth variants
+                        Arrays.asList(
+                                new TestReferenceBlockConcordanceVariant("C", 1, 1, 99)
```

Comment:
Similar issue to the GVCFWriter. Yes, a `VariantContext` can definitely hold all necessary information, but you can't just initialize it in one line. Instead you have to use a `VariantContextBuilder` and then add to that the genotypes and for that you have to use a `GenotypeBuilder`. My reasoning here was to be able to create test conditions easily and in a way that makes it absolutely clear what is happening, so that you can check the expected outcome by simply looking at the lines above.

Again, I agree that this is not the most efficient programming, but for the sake of clarity and robustness in test code maybe acceptable? What I did now is convert these four arguments (ALT allele, start, end, confidence) to an `Object[]`, which achieves the goal of having all information in one line and getting rid of the extra class. Let me know what you think

Reply:
Actually never mind, I moved the construction of a `VariantContext` to a static method, so you can create that object easily in the test condition definition, which is much more elegant than my original solution.